src/main/java/com/codegym/jrugotom5/service/AdvertService.java

```diff
@@ -64,4 +66,42 @@ public List<AdvertBasicInfoDTO> getByCategory(String category) {
         }
 
     }
+
+    @Transactional
+    public void deleteAdvertById(Long id) {
+        if (!advertRepository.existsById(id)) {
+            log.error("Advert with ID {} not found", id);
+            throw new EntityNotFoundException("Advert with %d ID not found".formatted(id));
+        }
+        advertRepository.deleteById(id);
+    }
+
+    @Transactional
+    public void deleteAdvertByTitle(String title) {
+        if (!advertRepository.existsByTitle(title)) {
+            log.error("Advert with title {} not found", title);
+            throw new EntityNotFoundException("Advert with title %s not found".formatted(title));
```

Comment:
Its better use custom exception for this purpose, something like AdvertDeletionException. Throw it here and do logging in GlobalExceptionHandler. Same goes for other endpoints including UserController.

Reply:
changes have been made